kpet/data.py

```diff
@@ -270,12 +303,22 @@ def matches(self, target):
         Returns:
             True if the case matches the target, False otherwise.
         """
-        return self.pattern.matches(target)
+        if target.sets is None or target.sets & self.sets:
+            # I think we don't need to set this to None once we remove sets
+            # from pattern
+            new_target = Target(trees=target.trees,
+                                arches=target.arches,
+                                components=target.components,
+                                sets=None,
+                                sources=target.sources)
+            return self.pattern.matches(new_target)
```

Comment:
I see what you did here. It would work, but I think it's a little too much of a hack and will bite us rather sooner than later. Especially in connection to FASTMOVING-1089.

What I think we could do for now, is to remove "sets" from `Target` altogether, and pass the sets to filter by to `run.Base.__init__()` alongside the `target`. Then we can do the set-based filtering in there explicitly.

Eventually, I think we'll need to remove the `matches()` method from suites and cases and make them passive, only housing the patterns. I feel like it's not really a job for each specific suite/case to decide whether to execute or not, it leads to a little to much complexity.

Reply:
I like the idea of having everything in Target and while this  sure is a hack, it just temporary for until we remove sets from pattern I think